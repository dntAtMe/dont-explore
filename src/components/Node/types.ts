import { NodeData } from "../../types";

export type NodeProps = {
  data: NodeData;
  onNodeClick: any;
};