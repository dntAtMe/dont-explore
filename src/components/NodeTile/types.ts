import { Node } from "../../utils/Node/Node";

export type NodeTileProps = {
  data: Node;
  onNodeClick: any;
};