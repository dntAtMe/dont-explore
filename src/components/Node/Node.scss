.node {
	
	background: none;
	width: 100px;
	height: 100px;

	&:hover {
		background-color: cyan;
	}

	&_name {
		width: 90%;
		word-wrap: break-word;
		overflow-x: hidden;
		overflow-y: hidden;
	}
}