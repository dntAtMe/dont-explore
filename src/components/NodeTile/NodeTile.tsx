import { NodeTileProps } from "./types";
import "./NodeTile.scss";

export const NodeTile = ({ data, onNodeClick }: NodeTileProps) => {

  const onClick = () => {
    onNodeClick(data);
  };

  return (
    <div onClick={onClick} data-testid="node" className='text-center node d-flex flex-column'>
      <i className=  { data.mode.type === 'directory' ? 'bi bi-folder' : 'bi bi-file-earmark' }></i>	
      <span className="node_name">{data.name}</span>
    </div>
  )
};

export default NodeTile;