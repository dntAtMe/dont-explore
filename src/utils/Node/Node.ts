import { FileMode, NodeData } from "../types";
import { parseFileMode } from "../utils";

export default class Node {
  _data: NodeData;

  constructor(data: NodeData) {
    this._data = data;
    console.log(data.mode);
  }

  get mode(): FileMode {
    return parseFileMode(this._data.mode);
  };

  get modification_timestamp() {
    return this._data.modification_date;
  };

  get name() {
    return this._data.name;
  };

  get size() {
    return this._data.size;
  };

  compareTo(other: Node): number {
    if (this.mode < other.mode) return 1;
    if (this.mode > other.mode) return -1;

    if (this.name < other.name) return 1;
    if (this.name > other.name) return -1;

    return 0;
  }

};