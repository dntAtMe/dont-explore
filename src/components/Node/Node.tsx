import { NodeProps } from "./types";
import "./Node.scss";

export const Node = ({ data, onNodeClick }: NodeProps) => {

	const onClick = () => {
		onNodeClick(data);
	};

	return (
		<div onClick={onClick} className='text-center node d-flex flex-column'>
			<i className=  { data.mode.startsWith('d') ? 'bi bi-folder' : 'bi bi-file-earmark' }></i>	
			<span className="node_name">{data.name}</span>
		</div>
	)
};

export default Node;