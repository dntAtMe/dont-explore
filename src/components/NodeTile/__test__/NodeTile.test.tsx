import { describe, it } from "@jest/globals";
import {fireEvent, render, screen} from '@testing-library/react'
import Node from "../../../utils/Node/Node";
import NodeTile from "../NodeTile";

describe('When NodeTile is rendered', () => {
  const node = new Node({
    mode: 'dr-sr-sr-x',
    modification_date: 100,
    name: 'directory',
    size: 120,
  });  

  it("it should render properly", () => {
    const { asFragment } = render(<NodeTile onNodeClick={jest.fn} data={node} />)
    expect(asFragment()).toMatchSnapshot();
  });

  it("Clicking it should call onClick function", () => {
    const fn = jest.fn();
    render(<NodeTile onNodeClick={fn} data={node} />)

    const tile = screen.getByTestId("node");
    fireEvent.click(tile);

    expect(fn).toBeCalled();
  });
});