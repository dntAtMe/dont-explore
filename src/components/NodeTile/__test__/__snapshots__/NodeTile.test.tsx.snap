// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When NodeTile is rendered it should render properly 1`] = `
<DocumentFragment>
  <div
    class="text-center node d-flex flex-column"
    data-testid="node"
  >
    <i
      class="bi bi-folder"
    />
    <span
      class="node_name"
    >
      directory
    </span>
  </div>
</DocumentFragment>
`;
